src/main/java/pl/betoncraft/betonquest/compatibility/placeholderapi/BetonQuestPlaceholder.java

```diff
@@ -76,6 +78,10 @@ public String onPlaceholderRequest(final Player player, final String identifier)
         final String placeholderIdentifier;
         final int index = identifier.indexOf(':');
         if (index == -1) {
+            if (Config.getDefaultPackage() == null) {
+                LogUtils.logThrowableIgnore(new QuestRuntimeException("A variable contains no package and default package is missing!"));
+                return "&cCould not resolve variable because it specifies no package and default package is missing.";
+            }
```

Comment:
Don't just create an exception to silently throw it again. Also don't set the value of the placeholder to the error message, but instead keep it empty and warn or error on the server console.
```suggestion
            if (Config.getDefaultPackage() == null) {
                LogUtils.getLogger().warn("Could not resolve placeholder value as variable '" + identifier + "' is specified without package and the default package is disabled.");
                return "";
            }
```

Reply:
I based my code on how invalid variable names are currently handled in BetonQuest:
https://github.com/BetonQuest/BetonQuest/blob/385a68cc30c2ac363bd7e8812f82f94fb2ac5317/src/main/java/pl/betoncraft/betonquest/BetonQuest.java#L1147-L1163

Silently throwing an exception is the intended way by wolf to include the stack trace in the default log. 

We should decide on a consistent way, sow should this behaviour be changed in master and master_1.12 or should I keep it how it is?

Reply:
The created and directly logged exception definitely wasn't intended, it probably happened during one of @Wolf2323's bulk fixing operations when we introduced PMD and FindBugs.
About the return value... Well, one could argue about it.

Reply:
Looking at the implementation of the variables: 

They all return an empty string and log the error if an error occured while resolving the value, so we should also use that behaviour for parsing to get things consistent.

I'll change that.